src/event-decoder/fees-collector/depositSwapFees.event.ts

```diff
@@ -0,0 +1,42 @@
+import { Address } from '@elrondnetwork/erdjs/out';
+import { RawEvent } from '../raw.event';
+import { FeesCollectorEventTopics } from "./fees-collector.event.topics";
+import { DepositSwapFeesEventType } from "./fees-collector.types";
+import BigNumber from "bignumber.js";
+import { RawEventType } from "../generic.types";
+
+export class DepositSwapFeesEvent extends RawEvent {
+    private decodedTopics: FeesCollectorEventTopics;
+
+    protected caller: Address | undefined;
+    private readonly paymentToken: string;
```

Comment:
You already have `paymentToken` and `currentWeek` into decoded topics. I don't think you need to duplicate them here

Reply:
done